4_Conceptos/4_Componente.md

```diff
@@ -0,0 +1,229 @@
+
+* Un componente agrupa el código de cada una de las capas de una [layered
+  architecture](/2_Tecnicas_y_herramientas/2_7_1_Layered_architecture.md), de
+  cada una de los filtros en una [pipes and filter
+  architecture](/2_Tecnicas_y_herramientas/2_7_2_Pipes_and_Filters_Architecture.md),
+  o como el núcleo y cada uno de los complementos en una [microkernel
+  architecture](/2_Tecnicas_y_herramientas/2_7_3_Microkernel_Architecture).
+
+* Un componente también puede ser un servicio en una [service based
+  architecture](/2_Tecnicas_y_herramientas/2_7_4_Service_Based_Architecture), un
+  micro—servicio en una [microservices
+  architecture](/2_Tecnicas_y_herramientas/2_7_8_Microservices_Architecture.md),
+  o en cualquier otro arquitectura distribuida.
+
+  En este caso cada componente ejecuta habitualmente en su propio espacio de
+  memoria y se comunica con los demás componentes mediante protocolos de red de
+  bajo nivel como TCP/IP o mediante mensajes de más alto nivel como llamados
+  REST o colas de mensajes, por ejemplo.
+
+### Componentes y su relación con la arquitectura y el rol de arquitecto
+
+Los componentes forman el bloque de construcción modular fundamental en la
+arquitectura, por eso es importante identificar correctamente los componentes de
+una arquitectura. De hecho, una de las decisiones principales que debe tomar
+un arquitecto se refiere a la partición de alto nivel de los componentes en la
+arquitectura[^1].
+
+Habitualmente el componente es el nivel más bajo con el que interactúa
+directamente un arquitecto —exceptuando tal vez métricas que afectan [atributos
+de calidad](/4_Conceptos/4_Atributos_de_calidad.md) en las bases del código. Las
+clases o funciones y procedimientos dentro de un componente son responsabilidad
+de los desarrolladores[^1].
+
+### Partición de la arquitectura en componentes
+
+En la medida de que los componentes representan un mecanismo genérico para
+contener código, el código se puede particionar en componentes de distintas
+maneras, cada una de ellas con diferentes compromisos en la satisfacción de los
+[atributos de calidad](./4_Atributos_de_calidad.md) de la arquitectura de
+software resultante. Sin embargo, el reparto de componentes de alto nivel merece
+un tratamiento especial, pues condiciona más que los demás la arquitectura del
+software.
+
+Existen dos grandes formas de comenzar a particionar los componentes en una
+arquitectura de software:
+
+* Partición técnica: El criterio para particionar los componentes de alto nivel
+  está basado en capacidades técnicas y es independiente del domino del
+  problema. Un ejemplo típico es cuando los componentes de alto nivel
+  corresponden a las capas de presentación, lógica de negocio, servicios y
+  persistencia, por ejemplo; de hecho, esta es la partición predeterminada en
+  diferentes organizaciones, frameworks, etc.
+
+  ![Partición técnica](/diagrams/Components_Technical_Partitioning.svg)
+
+  #### Figura 1: Partición técnica
```

Comment:
Entiendo que utilizamos h4 acá en vez de itálica para poder usarlo como hipervínculo luego. Igual pregunto: estamos ok con que las leyendas de las figuras sean h4 en vez de itálica?

Reply:
Tenés razón, por consistencia tendrían que estar en itálica; creo que quedó el
combo de las dos cosas, H4 e itálica. Agrego un TODO para cambiar
todos a este nuevo estilo y aclarlo en el CONTRIBUTING.md

Reply:
Bien. Pensando en voz alta me pregunto igual qué pasaría en casos donde queremos poner una imagen/figura ya estando dentro de un h4. Podemos dejar esto para solucionarlo como parte de ese TODO.